#3939 & #3940: Wallet signing phrase and welcome screens

alwx opened the pull request:
Fixes two issues:

#3939: Introducing wallet / welcome screen
#3940: Introducing wallet / signing phrase

alwx committed:
[#3939]: Introducing wallet / welcome screen;
[#3940]: Introducing wallet / signing phrase;
open-migrated-realm has been removed;

flexsurfer commented on src/status_im/data_store/realm/schemas/base/v1/account.cljs:
```diff
@@ -2,23 +2,25 @@
-                          :sharing-usage-data? {:type :bool :default false}
-                          :dev-mode?           {:type :bool :default false}
-                          :seed-backed-up?     {:type :bool :default false}}})
+             :properties {:address               :string
```
v1 in the release already

rasom commented:
@alwx any updates on this PR?

alwx commented:
@rasom coming today

alwx commented:
Somebody made tons of indentation changes, thus this branch required rebasing. I rebased it and fixed all issues mentioned here before. 
Can be checked now.

rasom commented:
> Somebody 

yeah, that was me

rasom commented:
there are also formatting errors
https://jenkins.status.im/blue/organizations/jenkins/status-react%2Fpull%20requests/detail/PR-4058/8/pipeline

<blockquote></blockquote>

alwx commented:
Fixed.

cammellos commented:
@alwx I had encountered some problems when changing schema version on develop as this code https://github.com/status-im/status-react/blob/3b8e2b7f75796813b5dea988597b3bc211f114f5/src/status_im/data_store/realm/core.cljs#L66 is triggered everytime I log in and out, resetting the database. Could you check if you have the same issue? (log in/log out, or restart the app, you loose all the chats). If so we probably want to remove that check with the PR.

flexsurfer commented:
@janherich please take a look at the comment above

janherich commented on ios/StatusIm.xcodeproj/project.pbxproj:
```diff
@@ -1566,6 +1587,13 @@
+			path = libRNSecureRandom.a;
+			remoteRef = B274C68220A069B400B55C41 /* PBXContainerItemProxy */;
+			sourceTree = BUILT_PRODUCTS_DIR;
+		};
```
Why are changes in this file necessary for this PR ?

janherich commented:
@flexsurfer @cammellos @alwx Yes, we should remove that check and include it conditionally whenever some future release will contain incompatible changes.
In my opinion, it should be done via flag and it should be simple check if schema version (`realm-version file-name encryption-key`) is different (`=`) from provided schema version `and` it's not `-1` (value for fresh install without any previous schema installed), if both conditions are false and flag is set, we just show the popup and delete the schema, if not, normal flow (migration to latest version) will proceed.
In case of any incompatible release in the future, it will be just matter turning the flag on (default will be false).
I can open PR with that logic, in the meantime you should remove that strange version check in any PR which increases the schema version.

alwx commented:
@janherich project.pbxproj changes have been removed

alwx commented:
@janherich @cammellos if I got it right, the problem should be solved by this commit (also included to this PR):
https://github.com/status-im/status-react/pull/4058/commits/4764313e35e9d3bed5a517b6d086d39fa3c82d33

cammellos commented:
@alwx code style check failed https://jenkins.status.im/blue/organizations/jenkins/status-react%2Fpull%20requests/detail/PR-4058/12/pipeline 

<blockquote></blockquote>

yevh-berdnyk commented:
Android: https://i.diawi.com/qpU67y
iOS: https://i.diawi.com/1Kq1EJ

Found issues
**Transaction phrase is changed after recovering an account in both cases if wallet was set up or it was not**
Reproduction:
- On device 1 create new account
- Navigate to Wallet and remember a transaction phrase
- Setup your wallet (you can skip this step and just remember the phrase)
- On device 2 recover created account
- Click Wallet
- Click Send Transaction
- Select a recipient and input some amount
- Click Sign transaction
- Transaction phrase is changed


<blockquote><div><strong><a href="https://i.diawi.com/qpU67y">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://i.diawi.com/1Kq1EJ">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

alwx commented:
This issue is not related to onboarding flow

yevh-berdnyk commented:
@alwx you right, I've checked on develop and wallet phrase is changed after recovering account.
I'll create a separate issue for it

goranjovic commented:
@alwx I see two commits in the PR. Is this intentional or do you want to squash them before merge?

flexsurfer commented:
@alwx could you resolve conflicts and squash commits please
